lib/ansible/plugins/action/nxos.py

```diff
@@ -41,8 +41,7 @@ def run(self, tmp=None, task_vars=None):
         self._config_module = True if module_name == 'nxos_config' else False
         socket_path = None
 
-        if (self._play_context.connection == 'httpapi' or self._task.args.get('provider', {}).get('transport') == 'nxapi') \
-                and module_name in ('nxos_file_copy', 'nxos_nxapi'):
+        if self._play_context.connection in ('httpapi', 'local') and module_name in ('nxos_file_copy', 'nxos_nxapi'):
```

Comment:
[This](https://github.com/ansible/ansible/pull/65876) PR deprecates `provider`. Hence, we would still have to honor provider.transport until it is removed.